spec/query_spec.cr

```diff
@@ -757,4 +757,18 @@ describe Avram::Query do
       end
     end
   end
+
+  describe "#reset_limit" do
+    it "resets the limit to nil so we can use aggregate functions again" do
+      users = UserQuery.new.limit(10)
+      users.reset_limit.select_count.should eq 0
+    end
+  end
+
+  describe "#reset_offset" do
+    it "resets the offset to nil so we can use aggregate functions again" do
+      users = UserQuery.new.offset(10)
+      users.reset_offset.select_count.should eq 0
```

Comment:
I’m not sure this is testing the reset offset or limit method since you could remove it and I believe the test would still pass. I think you may need to inspect the sql. And you may also want to do something like creating 2 users and check that bother users are returned

Reply:
`select_count` is an aggregate method, so if the query has an offset or a limit, it throws an error 
> Can't use aggregates (count, min, etc.) with limit or offset.

Removing the `reset_offset` makes this spec fail.